### PR TITLE
Adds snapshot publishing

### PR DESCRIPTION
**Pull request by JayShortway**
## Changes
- Adds a CircleCI config with a `deploy-snapshot` job that runs for all pushes to `main`. 
- Removes the NmcpPlugin, which I had to use in KobanKat. I had created my Maven Central account after a certain date which meant I could only publish via the [Central Portal](https://central.sonatype.org/register/central-portal/), and `com.vanniktech.maven.publish` did not have support for that at that time. (See https://github.com/vanniktech/gradle-maven-publish-plugin/issues/722.)
- Adjusts the coordinates and POM file to contain RevenueCat info instead of my personal info.
- Updates the klib abi dump, because the group changed (to `com.revenuecat.purchases`).

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+          command: |
+            set -euo pipefail
+            file_path="gradle/libs.versions.toml"
+            version=$(grep 'revenuecat-kmp = ' "$file_path" | awk -F ' = ' '{print $2}' | tr -d '"')
+            if [[ ! "$version" =~ -SNAPSHOT$ ]]; then
+              echo "$version is not a SNAPSHOT version. Exiting..."
+              exit 1
+            else
+              echo "$version is a SNAPSHOT version. Proceeding..."
+            fi
```
Comment (vegaro) [inline .circleci/config.yml:36]: not needed for now, but this should be moved to fastlane because it's easier to maintain and reuse. It's easier to use ruby than bash

**Review by vegaro — APPROVED** [review]

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+executors:
+  xcode15:
+    macos:
+      xcode: 15.4.0
+    resource_class: macos.m1.medium.gen1
+    environment:
+      HOMEBREW_NO_AUTO_UPDATE: 1
```
Comment (tonidero) [inline .circleci/config.yml:44]: Just curious why is this needed? Maybe we can put a comment explaining why.

**Review by tonidero — APPROVED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (JayShortway) [reply .circleci/config.yml:46]: Homebrew will check for updates every time you call `brew`. This disables that behavior so we don't have to wait for it. Everything we need is freshly installed, so that's already up to date. 
It's also a recommendation from CircleCI: https://circleci.com/docs/using-macos/#optimizing-homebrew

Added a comment with some details!

File: .circleci/config.yml (hunk 1 above)
Comment (JayShortway) [reply .circleci/config.yml:36]: Yes agreed! Although for now I don't even know how permanent this behavior is going to be, but will definitely move it to fastlane if it stays.
